File: Front-end/ProjetoSistemasDistribuidos/src/br/faj/sd/modelo/Pessoa.java
package br.faj.sd.modelo;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

import org.json.simple.JSONObject;

public class Pessoa extends Entidade
{
	private String cpf;
	private String rg;
	private String nome;
	private Date dataNascimento;
	private Sexo sexo;
	private boolean fuma;
	private boolean temFilhos;
	private Endereco endereco;
	private String telefone;
	private String email;
	private String foto;
	
	public enum Sexo
	{
		MASCULINO("Masculino"),
		FEMININO("Feminino");
		
		private String descricao;
		
		public String getDescricao()
		{
			return this.descricao;
		}
		
		private Sexo(String descricao)
		{
			this.descricao = descricao;
		}
	}

	public String getCpf()
	{
		return cpf;
	}

	public void setCpf(String cpf)
	{
		this.cpf = cpf;
	}

	public String getRg()
	{
		return rg;
	}

	public void setRg(String rg)
	{
		this.rg = rg;
	}

	public String getNome()
	{
		return nome;
	}

	public void setNome(String nome)
	{
		this.nome = nome;
	}

	public Date getDataNascimento()
	{
		return dataNascimento;
	}

	public void setDataNascimento(Date dataNascimento)
	{
		this.dataNascimento = dataNascimento;
	}

	public Sexo getSexo()
	{
		return sexo;
	}

	public void setSexo(Sexo sexo)
	{
		this.sexo = sexo;
	}

	public boolean isFuma()
	{
		return fuma;
	}

	public void setFuma(boolean fuma)
	{
		this.fuma = fuma;
	}

	public boolean isTemFilhos()
	{
		return temFilhos;
	}

	public void setTemFilhos(boolean temFilhos)
	{
		this.temFilhos = temFilhos;
	}

	public Endereco getEndereco()
	{
		return endereco;
	}

	public String getTelefone()
	{
		return telefone;
	}

	public void setTelefone(String telefone)
	{
		this.telefone = telefone;
	}

	public String getEmail()
	{
		return email;
	}

	public void setEmail(String email)
	{
		this.email = email;
	}

	public String getFoto()
	{
		return foto;
	}

	public void setFoto(String foto)
	{
		this.foto = foto;
	}
	
	public Pessoa()
	{
		this.cpf = "";
		this.rg = "";
		this.nome = "";
		this.dataNascimento = null;
		this.sexo = null;
		this.fuma = false;
		this.temFilhos = false;
		this.endereco = new Endereco();
		this.telefone = "";
		this.email = "";
		this.foto = "";
	}
	
	public Pessoa(JSONObject jsonObject) throws ClassCastException, ParseException
	{
		SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
		this.setCodigo(Integer.parseInt((String) jsonObject.get("codigo")));
		this.cpf = (String) jsonObject.get("cpf");
		this.rg = (String) jsonObject.get("rg");
		this.nome = (String) jsonObject.get("nome");
		this.dataNascimento = simpleDateFormat.parse((String) jsonObject.get("dataNascimento"));
		this.sexo = ((String) jsonObject.get("sexo")).equals("M") ? Sexo.MASCULINO : Sexo.FEMININO;
		this.fuma = ((String) jsonObject.get("fuma")).equals("S");
		this.temFilhos = ((String) jsonObject.get("temFilhos")).equals("S");
		this.endereco = new Endereco((JSONObject) jsonObject.get("endereco"));
		this.telefone = (String) jsonObject.get("telefone");
		this.email = (String) jsonObject.get("email");
		this.foto = (String) jsonObject.get("foto");
	}
	
	public String toString()
	{
		return this.nome;
	}
}

File: Back-end/modelo/entidade/Pessoa.php
<?php
    require_once __DIR__ . "/../../util/AtributoNaoEncontradoException.php";
    require_once __DIR__ . "/Entidade.php";
    require_once __DIR__ . "/Endereco.php";

    class Pessoa extends Entidade
    {
        private $cpf;
        private $rg;
        private $nome;
        private $dataNascimento;
        private $sexo;
        private $fuma;
        private $temFilhos;
        private $endereco;
        private $telefone;
        private $email;
        private $foto;

        public function __get($atributo)
        {
            switch ($atributo)
            {
                case "codigo": return $this -> codigo;
                case "cpf": return $this -> cpf;
                case "rg": return $this -> rg;
                case "nome": return $this -> nome;
                case "dataNascimento": return $this -> dataNascimento;
                case "sexo": return $this -> sexo;
                case "fuma": return $this -> fuma;
                case "temFilhos": return $this -> temFilhos;
                case "endereco": return $this -> endereco;
                case "telefone": return $this -> telefone;
                case "email": return $this -> email;
                case "foto": return $this -> foto;                
                default: throw new AtributoNaoEncontradoException($atributo);
            }
        }

        public function __set($atributo, $valor)
        {
            switch ($atributo)
            {
                case "codigo": $this -> codigo = $valor; break;
                case "cpf": $this -> cpf = $valor; break;
                case "rg": $this -> rg = $valor; break;
                case "nome": $this -> nome = $valor; break;
                case "dataNascimento": $this -> dataNascimento = $valor; break;
                case "sexo": $this -> sexo = $valor; break;
                case "fuma": $this -> fuma = $valor; break;
                case "temFilhos": $this -> temFilhos = $valor; break;
                case "telefone": $this -> telefone = $valor; break;
                case "email": $this -> email = $valor; break;
                case "foto": $this -> foto = $valor; break;
                default: throw new AtributoNaoEncontradoException($atributo);
            }
        }

        public function __construct()
        {
            parent::__construct();
            $this -> cpf = "";
            $this -> rg = "";
            $this -> nome = "";
            $this -> dataNascimento = "";
            $this -> sexo = "";
            $this -> fuma = false;
            $this -> temFilhos = false;
            $this -> endereco = new Endereco();
            $this -> telefone = "";
            $this -> email = "";
            $this -> foto = "";
        }

        public function jsonSerialize()
        {
            return array
            (
                "codigo" => $this -> codigo,
                "cpf" => $this -> cpf,
                "rg" => $this -> rg,
                "nome" => $this -> nome,
                "dataNascimento" => $this -> dataNascimento,
                "sexo" => $this -> sexo,
                "fuma" => $this -> fuma,
                "temFilhos" => $this -> temFilhos,
                "endereco" => $this -> endereco,
                "telefone" => $this -> telefone,
                "email" => $this -> email,
                "foto" => $this -> foto
            );
        }
    }
?>